persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -66,19 +65,12 @@ abstract class AbstractPersistentEntityRegistry(system: ActorSystem, injector: I
 
   private val registeredTypeNames = new ConcurrentHashMap[String, Class[_]]()
 
-  override def register[C, E, S](entityClass: Class[_ <: PersistentEntity[C, E, S]]): Unit = {
+  override def register(entityFactory: () => PersistentEntity[_, _, _]): Unit = {
```

Comment:
While I know that `() => PersistentEntity` is more correct, I wonder whether a by name param would look nicer.  In context, the application loader will either have this:

```scala
persistentEntityRegister.register(() => wire[MyEntity])
```

or this:

```scala
persistentEntityRegister.register(wire[MyEntity])
```

I think the big question is how important it is to make it clear to the reader that a factory is being passed.